src/otx/algo/segmentation/litehrnet.py

```diff
@@ -7,568 +7,42 @@
-            "__module.model.backbone.stage3.0/aten::add_/Add",
-            "__module.model.backbone.stage3.1/aten::add_/Add",
-        ]
-
-        return {
-            "ignored_scope": {
-                "patterns": ["__module.model.decode_head.aggregator/*"],
-                "names": ignored_scope_names,
-            },
-            "preset": "performance",
-        }
-
-
-LITEHRNET_VARIANTS = {
-    "LiteHRNet18": LiteHRNet18,
-    "LiteHRNetS": LiteHRNetS,
-    "LiteHRNetX": LiteHRNetX,
-}
-
-
-class OTXLiteHRNet(TorchVisionCompatibleModel):
+class LiteHRNet(OTXSegmentationModel):
     """LiteHRNet Model."""
 
-    def __init__(
-        self,
-        label_info: LabelInfoTypes,
-        input_size: tuple[int, int] = (512, 512),
-        optimizer: OptimizerCallable = DefaultOptimizerCallable,
-        scheduler: LRSchedulerCallable | LRSchedulerListCallable = DefaultSchedulerCallable,
-        metric: MetricCallable = SegmCallable,  # type: ignore[assignment]
-        torch_compile: bool = False,
-        backbone_configuration: dict[str, Any] | None = None,
-        decode_head_configuration: dict[str, Any] | None = None,
-        criterion_configuration: list[dict[str, Any]] | None = None,
-        export_image_configuration: dict[str, Any] | None = None,
-        name_base_model: str = "semantic_segmentation_model",
-    ):
-        super().__init__(
-            label_info=label_info,
-            input_size=input_size,
-            optimizer=optimizer,
-            scheduler=scheduler,
-            metric=metric,
-            torch_compile=torch_compile,
-            backbone_configuration=backbone_configuration,
-            decode_head_configuration=decode_head_configuration,
-            criterion_configuration=criterion_configuration,
-            export_image_configuration=export_image_configuration,
-            name_base_model=name_base_model,
-        )
+    AVAILABLE_MODEL_VERSIONS: ClassVar[list[str]] = [
+        "lite_hrnet_s",
+        "lite_hrnet_18",
+        "lite_hrnet_x",
+    ]
 
-    def _create_model(self) -> nn.Module:
-        litehrnet_model_class = LITEHRNET_VARIANTS[self.name_base_model]
-        # merge configurations with defaults overriding them
-        backbone_configuration = litehrnet_model_class.default_backbone_configuration | self.backbone_configuration
-        decode_head_configuration = (
-            litehrnet_model_class.default_decode_head_configuration | self.decode_head_configuration
-        )
-        # initialize backbones
-        backbone = LiteHRNet(**backbone_configuration)
-        decode_head = FCNHead(num_classes=self.num_classes, **decode_head_configuration)
+    def _build_model(self) -> nn.Module:
+        if self.model_version not in self.AVAILABLE_MODEL_VERSIONS:
+            msg = f"Model version {self.model_version} is not supported."
+            raise ValueError(msg)
 
-        return litehrnet_model_class(
+        backbone = LiteHRNetBackbone(self.model_version)
```

Comment:
`model_version` is used here but in the backbone initializer class `version` is used.
it'd be better to unify them.